### ISSUE TITLE
Font changes on its own within log-scale tick label

### ISSUE THREAD

**Issue by moble** [MakieOrg/Makie.jl#1296]
When using a font different from the default, and using a log scale, it is obvious that the font changes within the superscript of the tick labels.  Here's an MWE simplified from the [BeautifulMakie example](https://lazarusa.github.io/BeautifulMakie/ScattersLines/latexSample/):

```julia
using CairoMakie

fig = Figure(font = "CMU Serif")
ax = Axis(fig, yscale=log10)
lines!(0:1:50, x-> exp(-x))
fig[1,1] = ax
save("logscale.pdf", fig)  # Same result when using `display` in jupyter
save("logscale.png", fig)  # Same result when using `display` in jupyter
```
As you can see, the last digit in the exponents of the tick labels on the y scale are in the wrong font, and seemingly incorrectly aligned vertically:

[logscale.pdf](https://github.com/JuliaPlots/Makie.jl/files/7122615/logscale.pdf)

![logscale](https://user-images.githubusercontent.com/1470769/132370196-0f00ed02-6b90-4f5e-bb16-17f0a982667c.png)

(Note that the "last" digit is the only digit in the 10^0 and 10^-5  superscripts, but their heights are clearly inconsistent with the other superscripts, and at least the 5 is obviously sans serif.)

It seems that a workaround is to add the argument
```julia
ytickformat=ys->map(y->L"10^{%$(round(Int, log10(y)))}", ys)
```
to the `Axis` call, though that makes the superscripts a little lighter and harder to read.

Here's my version info:
```
(makie_logscale) pkg> status
      Status `/tmp/makie_logscale/Project.toml`
  [ee78f7c6] Makie v0.15.2

julia> versioninfo()
Julia Version 1.6.2
Commit 1b93d53fc4 (2021-07-14 15:36 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin18.7.0)
  CPU: Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, haswell)
```

**Comment by jonas-schulze** [comment]
Using e.g. LaTeX you can observe that some fonts distinguish between digits occuring in text and in math mode. Some digits decent below the [baseline](https://en.wikipedia.org/wiki/Baseline_(typography)) when in text mode but don't when in math mode. I suspect that tick labels are rendered like classical strings (cf https://github.com/JuliaPlots/Makie.jl/issues/1560#issuecomment-1063205916), i.e. in text mode only, and the exponents are "text mode superscripts". As a hotfix, maybe there is a different variant of your font that is identical up to this particular handling of digits.

**Comment by moble** [comment]
There is evidently a font-selection problem, but it's not just a baseline issue, or that the the base and exponent are in different fonts, it's that ***the font changes within the superscript!***

For example, if you zoom in on the $10^{-15}$ on the y axis, this is what you see:

![logscale-2](https://user-images.githubusercontent.com/1470769/157531482-b63eefe1-ec00-4e2d-84d6-f510cea4b26a.png)

The 1 and the 5 in the exponent are clearly different fonts — not just at different positions relative to the baseline.  In fact, if I crop that little section out of the PDF, `pdffonts` says that the $10^{-1}$ is in PTSerif-Regular for some reason, and the 5 is in DejaVuSans.  They're literally different fonts.

For reference, here is what `15` looks like in these three different fonts:

![test_mathfont_CMU](https://user-images.githubusercontent.com/1470769/157534363-717b9615-ebae-4862-94b5-7f87f0e7b490.png)

![test_mathfont_PTSerif](https://user-images.githubusercontent.com/1470769/157534608-4cf6e502-9425-4c56-a78f-f27f7472870c.png)

![test_mathfont_DejaVuSans](https://user-images.githubusercontent.com/1470769/157534518-d52b58d9-cf70-458d-8b65-b8c4da201586.png)

Note that the bottoms of 1 and 5 are aligned in all three fonts, and the results are consistent with what `pdffonts` is telling me about the output from CairoMakie.

**Comment by moble** [comment]
I've just tried this code again with a fresh julia project, in which the only dependency was `CairoMakie`, and got the same result, except that it now correctly uses CMU Serif for the `$10^{-1}$`.  Still DejaVu Sans for the 5, though.

**Comment by SimonDanisch** [comment]
At least for GLMakie, a fallback font gets used if a certain glyph can't be found in the font.
Since those are unicode superscripts which aren't supported equally well by all fonts, maybe it can't find the 5 in CMU Serif

**Comment by jonas-schulze** [comment]
@SimonDanisch, how did you find which glyphs are included in a particular font?

I was trying to find out which digit glyphs are included for Utopia, but could only find the "normal" digits. Could the issue be related to superscript 0,1,2 being in the [Latin-1 Supplement](https://unicodemap.org/range/2/Latin-1_Supplement/) group while 3-9 are in the [Superscripts and Subscripts](https://unicodemap.org/range/41/Superscripts_and_Subscripts/) range? Maybe, for the Latin-1 Supplement, some fonts know how to transform the default character set, while for the other group they just pass.

**Comment by SimonDanisch** [comment]
Yeah, haven't looked it up, but sounds about right that they're in different groups.

```julia
using FreeTypeAbstraction
font = FreeTypeAbstraction.findfont("Dejavu Sans")
FreeTypeAbstraction.FT_Get_Char_Index(font, '⁷') !== 0
```

**Comment by jonas-schulze** [comment]
Thanks! FYI, Utopia has glyphs for Latin-1 Supplement only (i.e. superscript 1,2,3; I fixed my comment above). I don't have CMU Serif installed, so I can't check.

**Comment by moble** [comment]
Ah, I didn't realize it was trying to use unicode to do superscripts.  Indeed, on my system at least, lualatex can only find superscripts in CMU Serif for 1, 2, and 3.  But it can actually create superscripts in the usual tex way.  So I guess the problem is to get ticklabels created using LaTeXStrings...

**Comment by moble** [comment]
This seems to have been resolved (possibly by #2321), at least as of Makie v0.19.7 and CairoMakie v0.10.7.

The relevant BeautifulMakie page is now [here](https://beautiful.makie.org/dev/examples/generated/2d/lines/line_latex/), and my MWE above would have to change as
```diff
-fig = Figure(font = "CMU Serif")
+fig = Figure(fonts = (;regular="CMU Serif"))
```
